internal/storage/storage.go

```diff
@@ -9,10 +9,38 @@ import (
 )
 
 type Storage interface {
+	Loader
+	Storer
+}
+
+type Loader interface {
 	Load(ctx context.Context, owner client.Object) (fs.FS, error)
+}
+
+type Storer interface {
 	Store(ctx context.Context, owner client.Object, bundle fs.FS) error
 	Delete(ctx context.Context, owner client.Object) error
 
 	http.Handler
 	URLFor(ctx context.Context, owner client.Object) (string, error)
 }
+
+type fallbackLoaderStorage struct {
+	Storage
+	fallbackLoader Loader
+}
+
+func WithFallbackLoader(s Storage, fallback Loader) Storage {
+	return &fallbackLoaderStorage{
+		Storage:        s,
+		fallbackLoader: fallback,
+	}
+}
+
+func (s *fallbackLoaderStorage) Load(ctx context.Context, owner client.Object) (fs.FS, error) {
+	fsys, err := s.Storage.Load(ctx, owner)
+	if err != nil {
```

Comment:
I figured it was best to fallback if _any_ error happens. That way we maximize the chance of success.

It would be up to the primary storage loader to retry, try alternate methods or whatever. Basically, a loader returning an error is essentially saying "I give up", in which case I think it makes sense for the fallback to give it a shot.

Reply:
I suppose we could make a `ErrSkipFallback` or something that loaders could use to tell any potential fallback loader not to bother. Seems unnecessary though.

Reply:
Yes, I think if we're saying if the primary storage loader fails for any reason, then use the fallback loader then this implementation makes sense without introducing some more refined error handling.